src/repositories/access_token.repository.ts

```diff
@@ -12,6 +12,8 @@ export interface OAuthTokenRepository {
 
   revoke?(accessToken: OAuthToken): Promise<void>;
 
+  revokeDescendantsOf?(authCodeId: string): Promise<void>;
```

Comment:
😂 this is what I was trying to do with the `revoke` method, making it optional as to not break backwards compatibility. This is great, thank you!